CMakeLists.txt

```diff
@@ -89,25 +89,18 @@ if(NOT MSVC)
 endif()
 
 if(Boost_FOUND)
-  get_property(SOCI_COMPILE_DEFINITIONS
-    DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    PROPERTY COMPILE_DEFINITIONS)
-
-  list(APPEND SOCI_COMPILE_DEFINITIONS "HAVE_BOOST=1")
-  list(APPEND SOCI_COMPILE_DEFINITIONS "BOOST_ALL_NO_LIB")
```

Comment:
Well, I know about static. It is second options. 
But current changes breaks Boost dynamic lib linkage which was fine before. 
@mloskot If it's ok - I'll be fine with this.